test/alternator/test_streams.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2019 ScyllaDB
+#
+# This file is part of Scylla.
+#
+# Scylla is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Scylla is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+
+# Tests for stream operations: ListStreams, DescribeStream, GetShardIterator.
+
+import pytest
+from botocore.exceptions import ClientError
+from util import list_tables, test_table_name, create_test_table, random_string
+
+stream_types = [ 'OLD_IMAGE', 'NEW_IMAGE', 'KEYS_ONLY', 'NEW_AND_OLD_IMAGES']
+
+def test_list_streams(dynamodb, dynamostreams):
+    for type in stream_types:
+        table = create_test_table(dynamodb,
```

Comment:
Right. Though in my testing it was not very slow at all...?

Reply:
Local tests are quite fast, the real slowness can be experienced when checking for regressions against actual DynamoDB, via our `--aws` switch:
```bash
pytest -vvv test/alternator --aws
```
for that you'll need to set up AWS credentials, all the details are in test/alternator/README.md.

Reply:
"slow" and "fast" are relative terms. A 30 second dtest is a "fast" dtest, while in 30 seconds we run the entire Alternator test suite locally ;-)
Anyway, creating a table locally isn't too slow - it takes a (not small) fraction of a second - but it is definitely the *slowest* part of the tests, so we try hard to reuse the same table in many tests. As Piotr said, for remote tests over DynamoDB, creating a table can be *really* slow - often many seconds - but on the other hand we only run these tests on DynamoDB rarely, when developing tests.
Another benefit of the "fixture" idea of reusing the same test table for many tests is that it makes it really easy and fast to have even dozens of small tests that each takes milliseconds to run. You don't need one huge test that verifies 10 different things just because of the big table-creation overhead - instead you have a fixture which creates the table and 10 small tests which each does a small test on it. The small test functions make it much easier to understand what is actually being tested, and when something fails, what failed.
If you want I can give you an even better introduction to pytest and fixtures - but let's take it on the mailing list.